R/i18n.R

```diff
@@ -150,10 +150,73 @@ i18n_set_language_option <- function(language = NULL) {
   }
 
   knitr::opts_knit$set(tutorial.language = language)
+  i18n_setenv_language(language)
 
   invisible(current)
 }
 
+i18n_setenv_language <- function(lang) {
+  lang <- i18n_determine_base_r_language(lang)
+
+  old_lang <- Sys.getenv("LANGUAGE", unset = "en")
+  old_text <- gettext("subscript out of bounds", domain = "R")
+
+  Sys.setenv("LANGUAGE" = lang)
+
+  new_lang <- Sys.getenv("LANGUAGE", unset = "en")
+  new_text <- gettext("subscript out of bounds", domain = "R")
+
+  if (old_lang != new_lang && old_text == new_text) {
```

Comment:
All four things are almost certainly length-1 but this covers our bases

```suggestion
  if (!identical(old_lang, new_lang) && identical(old_text, new_text)) {
```